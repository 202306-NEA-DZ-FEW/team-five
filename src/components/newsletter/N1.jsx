import Image from "next/image";
import React from "react";

const N1 = () => {
    return (
        <div>
            <div className='flex flex-row'>
                <div className='hidden md:block'>
                    <Image
                        src='/heart.png'
                        width={200}
                        height={200}
                        alt='no image'
                    />
                </div>
                <div className='p-10 text-[#212B36]'>
                    <b>join our newsletter to learn more</b>
                    <h1>We need you to stop hunger!</h1>

                    <form
                        action='https://formsubmit.co/sidisaidmel@gmail.com'
                        method='POST'
                        className='mt-10'
                    >
                        <input
                            type='text'
                            name='name'
                            placeholder='Name'
                            required
                            className='pl-2 mr-5 h-9 border rounded-sm mb-3'
                        />
                        <input
                            type='email'
                            name='email'
                            placeholder='E-mail'
                            required
                            className='pl-2 mr-5 h-9 border rounded-sm'
                        />
                        <input
                            type='hidden'
                            name='_subject'
                            value='New submission!'
                        />
                        <button
                            type='submit'
                            className='bg-[#20DC49] text-white px-16  font-semibold mt-3 h-9 rounded-sm w-48'
                        >
                            Send
                        </button>
                        <input
                            type='hidden'
                            name='_autoresponse'
                            value='Successfully subscribed'
                        ></input>
                    </form>
                </div>
            </div>
        </div>
    );
};

export default N1;
