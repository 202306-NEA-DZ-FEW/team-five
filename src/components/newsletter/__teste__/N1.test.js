import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

import N1 from "../N1";

jest.mock("next/router", () => ({
    useRouter: () => ({
        route: "/",
        pathname: "",
        query: {},
        asPath: "",
    }),
}));

jest.mock("react-i18next", () => ({
    useTranslation: () => ({ t: (key) => key, i18n: { language: "en" } }),
}));

const mockStore = configureMockStore([]);
const store = mockStore({
    userInfo: null,
});

describe("Button Component", () => {
    it("should render correctly", () => {
        const component = renderer.create(
            <Provider store={store}>
                <N1 />
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
