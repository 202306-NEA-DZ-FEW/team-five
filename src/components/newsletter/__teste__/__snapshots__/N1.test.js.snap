// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
<div
  className=""
>
  <div
    className="grid grid-rows-3 grid-flow-col gap-4"
  >
    <div
      className="row-span-3 ..."
    >
      <img
        alt="no image"
        data-nimg="1"
        decoding="async"
        height={200}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fheart.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fheart.png&w=384&q=75 1x, /_next/image?url=%2Fheart.png&w=640&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={300}
      />
    </div>
    <div
      className="col-span-2 text-xl  ..."
    >
      <b>
        join our newsletter to learn more
      </b>
      <h1>
        We need you to stop hunger!
      </h1>
    </div>
    <div
      class="row-span-2 col-span-2 mt-5 ..."
    >
      <form
        action="https://formsubmit.co/sidisaidmel@gmail.com"
        method="POST"
      >
        <input
          name="name"
          placeholder="name"
          required={true}
          style={
            Object {
              "border": "1px solid black",
              "textAlign": "center",
            }
          }
          type="text"
        />
        <input
          name="email"
          placeholder="email"
          required={true}
          style={
            Object {
              "border": "1px solid black",
              "marginLeft": "20px",
              "textAlign": "center",
            }
          }
          type="email"
        />
        <input
          name="_subject"
          type="hidden"
          value="New submission!"
        />
        <button
          style={
            Object {
              "backgroundColor": "green",
              "color": "white",
              "marginLeft": "10px",
              "padding": "2px",
              "width": "200px",
            }
          }
          type="submit"
        >
          Send
        </button>
        <input
          name="_autoresponse"
          type="hidden"
          value="Successfully subscribed"
        />
      </form>
    </div>
  </div>
</div>
`;
